TrafficCapture/captureKafkaOffloader/src/main/java/org/opensearch/migrations/trafficcapture/kafkaoffloader/tracing/KafkaRecordContext.java

```diff
@@ -0,0 +1,62 @@
+package org.opensearch.migrations.trafficcapture.kafkaoffloader.tracing;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.metrics.Meter;
+import lombok.Getter;
+import lombok.NonNull;
+import org.opensearch.migrations.tracing.BaseNestedSpanContext;
+import org.opensearch.migrations.tracing.CommonScopedMetricInstruments;
+import org.opensearch.migrations.tracing.IScopedInstrumentationAttributes;
+import org.opensearch.migrations.tracing.commoncontexts.IConnectionContext;
+
+public class KafkaRecordContext extends
+        BaseNestedSpanContext<IRootKafkaOffloaderContext, IConnectionContext>
+        implements IScopedInstrumentationAttributes {
+    public static final String ACTIVITY_NAME = "kafkaCommit";
+
+    static final AttributeKey<String> TOPIC_ATTR = AttributeKey.stringKey("topic");
+    static final AttributeKey<String> RECORD_ID_ATTR = AttributeKey.stringKey("recordId");
+    static final AttributeKey<Long> RECORD_SIZE_ATTR = AttributeKey.longKey("recordSize");
+
+    @Getter
+    public final String topic;
+    @Getter
+    public final String recordId;
+    @Getter
+    public final int recordSize;
+
+    public KafkaRecordContext(IRootKafkaOffloaderContext rootScope, IConnectionContext enclosingScope,
+                              String topic, String recordId, int recordSize) {
+        super(rootScope, enclosingScope);
+        this.topic = topic;
+        this.recordId = recordId;
+        this.recordSize = recordSize;
+        initializeSpan();
+    }
+
+    public static class MetricInstruments extends CommonScopedMetricInstruments {
+        private MetricInstruments(Meter meter, String activityName) {
+            super(meter, activityName);
+        }
+    }
+
+    public static @NonNull MetricInstruments makeMetrics(Meter meter) {
+        return new MetricInstruments(meter, ACTIVITY_NAME);
+    }
+
+    @Override
+    public @NonNull MetricInstruments getMetrics() {
+        return getRootInstrumentationScope().getKafkaOffloadingInstruments();
+    }
+
+    @Override
+    public String getActivityName() { return "stream_flush_called"; }
+
+    @Override
+    public AttributesBuilder fillAttributes(AttributesBuilder builder) {
```

Comment:
Discussed offline, my preference is to chain our extending interface's fillAttributes so that our metrics are additive. May require adding an override to in between interfaces to satisfy java rules.

Also, call the attributeBuilder puts after the super call.

Reply:
This is cumbersome to do with Java, but the compiler does provide some protections.  For a default method defined within a hierarchy of interfaces, I can specify the name of any interface _after_ the deepest default implementation and those will compile without error.  **However**, if refer to say `IInstrumentationAttributes.super.fillAttributes()` but then implement fillAttributes() in IWithStartTimeAndAttributes, the compiler will complain with `java: not an enclosing class: org.opensearch.migrations.tracing.IWithStartTimeAndAttributes`